twoliter/src/cargo_make.rs

```diff
@@ -0,0 +1,213 @@
+use std::path::PathBuf;
+
+use anyhow::{bail, Result};
+use log::trace;
+use tokio::process::Command;
+
+use crate::common::exec;
+use crate::docker::ImageArchUri;
+use crate::project::Project;
+
+fn require_sdk(project: &Project, arch: &str) -> Result<(ImageArchUri, ImageArchUri)> {
+    match (project.sdk(arch), project.toolchain(arch)) {
+        (Some(s), Some(t)) => Ok((s, t)),
+        _ => bail!(
+            "When using twoliter make, it is required that the SDK and toolchain be specified in \
+            Twoliter.toml"
+        ),
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct CargoMake {
+    makefile_path: Option<PathBuf>,
+    project_dir: Option<PathBuf>,
+    args: Vec<String>,
+}
+
+impl CargoMake {
+    pub fn new<S>(project: &Project, arch: S) -> Result<Self>
+    where
+        S: Into<String>,
+    {
+        let (sdk, toolchain) = require_sdk(project, &arch.into())?;
+        Ok(Self::default()
+            .env("TLPRIVATE_SDK_IMAGE", sdk)
+            .env("TLPRIVATE_TOOLCHAIN", toolchain))
```

Comment:
I'm thinking about whether we should just hang on to `&'a Project` and do things like this during the exec call. I'm also slightly concerned about lack of clarity regarding what args could be passed during `exec` and which are managed by this object.

These are not huge concerns. Just some things to consider and change if it feels like we can do better.